src/main/java/appeng/parts/p2p/PartP2PFluids.java

```diff
@@ -252,15 +253,15 @@ public FluidStack drain( int maxDrain, boolean doDrain )
 
 		if( s == null )
 		{
-			DEPTH.set( s = new LinkedList<>() );
+			DEPTH.set( s = new ArrayDeque<>() );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment out of this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)